FirebaseInstallations/Source/Library/FIRSecureStorage.m

```diff
@@ -26,28 +26,108 @@
 #import "FIRInstallationsErrorUtil.h"
 
 @interface FIRSecureStorage ()
-@property(nonatomic, strong) dispatch_queue_t keychainQueue;
+@property(nonatomic, readonly) dispatch_queue_t keychainQueue;
+@property(nonatomic, readonly) dispatch_queue_t inMemoryCacheQueue;
 @property(nonatomic, readonly) NSString *service;
+@property(nonatomic, readonly) NSCache<NSString *, id<NSSecureCoding>> *inMemoryCache;
 @end
 
 @implementation FIRSecureStorage
 
 - (instancetype)init {
-  return [self initWithService:@"com.firebase.FIRInstallations.installations"];
+  NSCache *cache = [[NSCache alloc] init];
+  // Cache up to 5 installations.
+  cache.countLimit = 5;
```

Comment:
what is this for?

Reply:
It is a guard against a memory consumption growth. There will be no more than 5 items in the cache. If a 6th is added the oldest accessed item will be removed from the cache. 

It is probably not too important for our use case so I may remove it if it is confusing. I consider moving FIRSecureStorage to GoogleUtilities later on to provide a universal way of storing data to the keychain. In this case it may be more relevant and I can re-introduce it.

Reply:
+1 on making it a universal way handling keychain.